parcels/fieldset.py

```diff
@@ -140,8 +120,7 @@ def add_field(self, field: Field, name: str | None = None):
         if hasattr(self, name):  # check if Field with same name already exists when adding new Field
             raise RuntimeError(f"FieldSet already has a Field with name '{name}'")
         else:
-            setattr(self, name, field)
```

Comment:
This `if hasattr...` block can be removed (duplication of check above), and the else statement can be de-indented